src/tokenizer.ts

```diff
@@ -1467,6 +1486,12 @@ export class Tokenizer extends DiagnosticEmitter {
         }
         sepEnd = pos + 1;
       } else {
+        if (!isEndOfLiteral(c)) {
```

Comment:
Perhaps the checks can be moved to a place with less duplication. Perhaps into `testInteger`? Might have to extend it with more cases, though, hmm.

Reply:
Unfortunately it required a lot of refactorings and much more complex logic for `readInteger`